src/Agent.Sdk/SecretMasking/LoggedSecretMasker.cs

```diff
@@ -137,6 +144,24 @@ public string MaskSecrets(string input)
             return this._secretMasker.MaskSecrets(input);
         }
 
-        ISecretMaskerVSO ISecretMaskerVSO.Clone() => this.Clone();
+        ISecretMasker ISecretMasker.Clone() => this.Clone();
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                if (_secretMasker is IDisposable disposable)
```

Comment:
Looks good!  (I had to look this up that is operator returns false if disposable is null)

https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/expressions#121212-the-is-operator